doc/sphinxdoc/generate_reference.py

```diff
@@ -147,6 +147,33 @@ def related_tutorials(algo_doc):
     ## 2. First Optimization - Create an index from each algorithm to file.
     ## 3. Use this mapping to go from algorithm to list of tutorial files in O(1) time.
 
+def source_links(algo_doc):
+    """
+    Get the source cpp and header links for the algorithm
+    """
+    # Source Directory and URL Prefix
+    SOURCE_PATH = '../../src/algorithms'
+    URL_PREFIX = 'https://github.com/MTG/essentia/tree/master/'
```

Comment:
Let's use prefix based on the source code used to build the documentation. Use the output of `git describe --long` instead of `master``.